### PR TITLE
[docs] distributed training

### PR DESCRIPTION
**Pull request by stevhliu**
- removes "Number of accelerators" section from "Accelerator selection" guide since this is probably pretty commonly known
- add a new "DDP" guide
- refactored "Accelerate" guide with a more focused overview of what it is and how to configure it (only covers `Trainer` now instead of raw PyTorch loop). this guide links out to more detailed FSDP/DDP guides for full config-specific settings. also documents some `AcceleratorConfig` settings
- refactored "FSDP" guide with cleaner table and `auto_wrap` explanation and detailed explanation of what the config settings are

### CODE REVIEW

File: docs/source/en/fsdp.md [hunk 1]
```diff
@@ -14,132 +14,124 @@ rendered properly in your Markdown viewer.
...
-tpu_use_sudo: false
-use_cpu: false
+</hfoption>
+<hfoption id="FSDP config file">
+
+Pass an FSDP config file to [`fsdp_config`]. All fields are optional except for the sharding strategy in `fsdp`.
+
+```json
+{
+  "version": 1,
```
Comment (SunMarc) [inline docs/source/en/fsdp.md]: Let's set it to 2. We will most likely switch to fsdpv2 now and deprecate the old path soon.

File: docs/source/en/fsdp.md [hunk 2]
```diff
@@ -14,132 +14,124 @@ rendered properly in your Markdown viewer.
...
+|---|---|
+| `full_shard` | shard parameters, gradients, and optimizer states |
+| `shard_grad_op` | shard gradients and optimizer states |
+| `no_shard` | DDP |
+| `hybrid_shard` | full shard within a node, replicate across nodes |
+| `hybrid_shard_zero2` | shard gradients and optimizer states within a node, replicate across nodes |
+| `offload` | CPU offload (combine with `full_shard` or `shard_grad_op`) |
 
-The section below discusses some of the more important FSDP configuration options. Learn more about other available options in the [fsdp_config](https://hf.co/docs/transformers/main_classes/trainer#transformers.TrainingArguments.fsdp_config) parameter.
+Always combine a sharding strategy with `auto_wrap` to enable the auto-wrapping policy like `fsdp="full_shard auto_wrap"`. Without `auto_wrap`, the entire model is one FSDP unit and you lose the memory benefit of sharding.
```
Comment (SunMarc) [inline docs/source/en/fsdp.md]: Let's update the docs for fsdpv2 args only. I feel like this could be a nicer transition instead of keeping the old arguments. We don't have this arg anymore. We only have `reshard_after_forward` now.

File: docs/source/en/accelerate.md [hunk 3]
```diff
@@ -16,150 +16,111 @@ rendered properly in your Markdown viewer.
...
   fsdp_backward_prefetch_policy: BACKWARD_PRE
-  fsdp_forward_prefetch: false
   fsdp_cpu_ram_efficient_loading: true
   fsdp_offload_params: false
   fsdp_sharding_strategy: FULL_SHARD
   fsdp_state_dict_type: SHARDED_STATE_DICT
   fsdp_sync_module_states: true
-  fsdp_transformer_layer_cls_to_wrap: BertLayer
+  fsdp_transformer_layer_cls_to_wrap: LlamaDecoderLayer
   fsdp_use_orig_params: true
```
Comment (SunMarc) [inline docs/source/en/accelerate.md]: update this for fsdpv2 spec

File: docs/source/en/fsdp.md [hunk 4]
```diff
@@ -14,132 +14,124 @@ rendered properly in your Markdown viewer.
...
+```py
+from transformers import TrainingArguments
 
-```bash
-accelerate launch --fsdp="full shard" --fsdp_config="path/to/fsdp_config/" my-training-script.py
+TrainingArguments(
+    ...,
+    fsdp="full_shard auto_wrap",
+    fsdp_config="path/to/fsdp.json",
+)
```
Comment (SunMarc) [inline docs/source/en/fsdp.md:118]: for now, let's ask the user to pass fsdp=True. In the future, they won't have to pass this arg anymore.

File: docs/source/en/accelerate.md [hunk 5]
```diff
@@ -16,150 +16,111 @@ rendered properly in your Markdown viewer.
...
-    args=training_args,
-    train_dataset=dataset["train"],
-    eval_dataset=dataset["test"],
-    processing_class=tokenizer,
-    data_collator=data_collator,
-    compute_metrics=compute_metrics,
+TrainingArguments(
+    ...,
+    fsdp="full_shard auto_wrap",
+    fsdp_config="path/to/fsdp.json",
```
Comment (SunMarc) [inline docs/source/en/accelerate.md:81]: same here

**Review by SunMarc — COMMENTED** [review]
Thanks ! Left a comment but overall, we will move on to FSDPv2 very soon. I'm thinking about changing the default to that soon so we should update the docs to reflect fsdpv2 arguments.

**Comment by stevhliu** [comment]
all the args should be updated to fsdpv2 now, lmk if there's anything else @SunMarc :)

**Review by SunMarc — APPROVED** [review]
Thanks for going through fsdpv2 docs, this is much better !
